docs/source/developer/architecture.rst

```diff
@@ -431,14 +431,177 @@ ufunc-like function that is inlined into a single loop that only
 allocates a single result array.
 
 
+.. _`parallel-accelerator`:
+
+Stage 6b: Perform Automatic Parallelization
+-------------------------------------------
+
+This pass is only performed if the parallel option in the :func:`~numba.jit`
+decorator is set to True.  This pass find parallelism implicit in the 
+semantics of operations in the Numba IR and replaces those operations
+with explicitly parallel representations of those operations using a
+special `parfor` operator.  Then, optimizations are performed to maximize
+the number of parfors that are adjacent to each other such that they can
+then be fused together into one parfor that takes only one pass over the
+data and will thus typically have better cache performance.  Finally,
+during lowering, these parfor operators are converted to a form similar
+to guvectorize to implement the actual parallelism.  All these processes
+are described in more detail in the following paragraphs.
+
+The automatic parallelization pass has a number of sub-passes.
+
+1. CFG simplification - Sometimes Numba IR will contain chains of blocks
```

Comment:
Without blank lines between, it puts it all in one paragraph for me.  Indent didn't help either.  Please help here.